### PR TITLE
755871123 - AAUser on the CDB page, when I click on a dropdown at the top of the table, the dropdown options will appear below the cell and vice versa

### PR DESCRIPTION
**Pull request by elio-stalteri**

### CODE REVIEW

File: src/components/organism/Table/CellRenderers/SelectRenderer.js [hunk 1]
```diff
@@ -19,46 +30,63 @@ export class Select extends React.PureComponent {
...
-  handleChange = selectedOption => {
+  handleChange = (selectedOption) => {
     this.setState({ value: selectedOption ? selectedOption.value : selectedOption }, () =>
       this.props.api.stopEditing(false),
     )
   }
 
   getValue = () => this.state.value
 
+  calculateDirection = () => {
```
Comment (alexwong-crc) [inline src/components/organism/Table/CellRenderers/SelectRenderer.js:70]: this is really easy to read and understand. Nice.

File: src/components/organism/Table/CellRenderers/SelectRenderer.js [hunk 2]
```diff
@@ -6,8 +6,19 @@ import ReactSelect from "react-select"
 import styles from "../../../atoms/Select/SelectStyles"
 
 export class Select extends React.PureComponent {
+  state = {
+    menuOpen: false,
+  }
+
+  componentDidMount() {
+    // this is needed due to a bug of react-select that doesn't consider menuPlacement before is fully rendered
```
Comment (alexwong-crc) [inline src/components/organism/Table/CellRenderers/SelectRenderer.js:14]: You said there was an open issue for this. Can you provide a link?

File: src/components/organism/Table/CellRenderers/SelectRenderer.js [hunk 3]
```diff
@@ -19,46 +30,63 @@ export class Select extends React.PureComponent {
...
   }
 
   getValue = () => this.state.value
 
+  calculateDirection = () => {
+    // get the distance from the top of the body without considering scroll
+    const distanceFromTop = this.props.node.rowTop
+    // get the body html element
+    const tableBodyElement = this.props.agGridReact.eGridDiv.querySelector(".ag-body-viewport")
+    // get the ammount scrolled inside the table
```
Comment (alexwong-crc) [inline src/components/organism/Table/CellRenderers/SelectRenderer.js:75]: typo `amount`

File: src/components/organism/Table/CellRenderers/SelectRenderer.js [hunk 4]
```diff
@@ -19,46 +30,63 @@ export class Select extends React.PureComponent {
...
+    const scrolledDistanceFromTop = tableBodyElement.scrollTop
+    // get the height of the table without considering the content
+    const tableBodyHeight = tableBodyElement.getBoundingClientRect().height
+    // calculate the distance of the row from the top of the table considering the scroll
+    const RelativeDistance = distanceFromTop - scrolledDistanceFromTop
```
Comment (alexwong-crc) [inline src/components/organism/Table/CellRenderers/SelectRenderer.js:80]: lower case r please.

Comment (elio-stalteri) [reply src/components/organism/Table/CellRenderers/SelectRenderer.js:80]: good spot

File: src/components/organism/Table/CellRenderers/SelectRenderer.js (hunk 2 above)
Comment (elio-stalteri) [reply src/components/organism/Table/CellRenderers/SelectRenderer.js:14]: done

**Review by alexwong-crc — APPROVED** [review]

**Pull request merged**